app/assets/src/components/ResultsFolder.jsx

```diff
@@ -1,12 +1,123 @@
 import React from "react";
 import Divider from "./layout/Divider";
 
+class OutputFile extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.file = props.file;
+  }
+
+  download(url) {
+    if (url) {
+      location.href = `${url}`;
+    }
+  }
+
+  render() {
+    return (
+      <tr
+        className={`${this.file.url ? "" : "disabled-"}file-link`}
+        onClick={this.download.bind(this, this.file.url)}
+      >
+        <td>
+          <i className="fa fa-file" />
+          {this.file["display_name"]}
+          <span className="size-tag"> -- {this.file["size"]}</span>
+        </td>
+      </tr>
+    );
+  }
+}
+
+class ConfigFile extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.stageDagJson = props.stageDagJson;
+  }
+
+  download_string2file(str) {
```

Comment:
same with these functions for camelCase and CSS Modules

Reply:
well, I guess the CSS is not really new

Reply:
yeah the CSS is not new and the download function is also not new.  Just broke single ResultsFolder component into some smaller pieces.
I will rename `downloadString2file`. This whole page is going to be redesigned and refactored in the future